### PR TITLE
Prepare to refactor Attribution Reporting helper libraries

### PR DESCRIPTION
**Pull request by siddharth-sahoo**
# Description

Note: This PR is for the unified branch.

Prepare to refactor Attribution Reporting helper libraries.

### Changes to `ad-tech` service

* Refactor `attribution-reporting-helper` such that it no longer interacts with the HTTP request/response directly. Instead this library can return the Attribution Reporting source and trigger registration headers for a given request context. The request to register sources and triggers are `GET` requests, so the request context is merely the query parameters in the URL. Source registration, additionally, needs the parsed structured request header indicating registration eligibility.
  * Accordingly, adjust `attribution-reporting-router` to handle permutative scenarios where request may not be eligible for attribution reporting or the header may be malformed. Repeat the process for `report-router` where event-level Protected Audience reports may be eligible for registering attribution sources _(ad beacon requests via Fenced Frames Ads Reporting)_.
* Add a placeholder comment to implement verbose debug reports for Attribution Reporting.
* Make logging statements in `arapi.ts` slightly more descriptive.
* Rename `template-utils` to `common-utils`.
  * Split `getAdTemplateVariables()` into 2 functions: `getContextualAdTemplateVariables()` and `getInterestGroupAdTemplateVariables()` for readability.
  * Convert `getJavaScriptTemplateVariables()` into a constant: `JAVASCRIPT_TEMPLATE_VARIABLES`.
  * Implement a new utility function: `getStructuredObject()`, which converts a generic object with `any` typed keys and values into a stringified structured object.
* Temporarily disable redirection of attribution reporting source registration headers for Protected Audience ads to make debugging easier. This is done by removing the `redirect` query parameter from ad beacon URLs in bidding logic.
* Merge `dsp/index` and `ssp/index` into a common `index` page for the entire `ad-tech` service. This page is not used in tests, though users are added to an interest group if they visit this page. This is a corner case where ad-tech itself is the advertiser.

## Affected services

- [ ] Home
- [ ] News
- [ ] Shop
- [ ] Travel
- [x] Ad-tech

Other:

### CODE REVIEW

File: services/ad-tech/src/routes/common/report-router.ts [hunk 1]
```diff
@@ -30,6 +37,29 @@ import {getTemplateVariables} from '../../lib/template-utils.js';
  */
 export const ReportRouter = express.Router();
 
+/** Sets ARA source registration headers if request is eligible. */
+const setAttributionReportingHeadersIfEligible = (
```
Comment (chasinandrew) [inline services/ad-tech/src/routes/common/report-router.ts:41]: Can this just be `setAttributionReportingHeaders`? Then can we add an else statement to catch if the eligibility header is not included in the request? So the "if eligible" piece is inferred and caught if the header isn't provided

**Review by chasinandrew — APPROVED** [review]
One nitpick on naming that can be applied to other methods with `IfNeeded` suffix, otherwise LGTM

File: services/ad-tech/src/routes/common/report-router.ts (hunk 1 above)
Comment (siddharth-sahoo) [reply services/ad-tech/src/routes/common/report-router.ts]: Ack, done.
